File: 1588. Sum of All Odd Length Subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        ans = 0
        
        for i in range(len(arr)):
            left, right = i/2, (len(arr)-i-1)/2
            even = (math.floor(left)+1)*(math.floor(right)+1)
            odd = math.ceil(left)*math.ceil(right)
            ans += (even+odd)*arr[i]
        
        return ans

File: 1588. Sum of All Odd Length Subarrays.js
/**
 * @param {number[]} arr
 * @return {number}
 */
 var sumOddLengthSubarrays = function(arr) {
    let ans = 0;

    for(let i=0;i<arr.length;i++){
        let [left, right] = [i/2, (arr.length-i-1)/2];
        let [even, odd] = [(Math.floor(left)+1)*(Math.floor(right)+1), Math.ceil(left)*Math.ceil(right)];
        ans += (even+odd)*arr[i];
    }
    return ans;
};